irpc-iroh/examples/auth.rs

```diff
@@ -23,7 +23,7 @@ async fn remote() -> Result<()> {
         let router = Router::builder(endpoint.clone())
             .accept(StorageServer::ALPN, server.clone())
             .spawn();
-        let addr = endpoint.node_addr().await?;
```

Comment:
Not a fan of this. But unfortunately it is not possible to implement IntoFuture for a trait.